### PR TITLE
Fix third_party coverage issue and coverage link.

### LINKED ISSUE #902: Third party included in code coverage / fix coverage link
It seems like the third party libs are included in the coverage of our code base. PR #893 apparently decreased by 1.4%. If you look at the [diff](https://ares.epic.hpi.uni-potsdam.de/jenkins/job/Hyrise/job/hyrise/job/lawben%252Fjson_benchmark/8//artifact/coverage_diff.html), it looks like most of this is caused by `third_party/json/json.hpp`. 

I assume this is not expected behaviour?

### PR DESCRIPTION
**Pull request by Bensk1**
Fixes #902. This removes the third_party, test and benchmark directories from the coverage report and diff. Also, it fixes the link pointing from our main readme to the coverage report.

### CODE REVIEW

File: DEPENDENCIES.md [hunk 1]
```diff
@@ -27,7 +27,8 @@
 
 ## Dependencies that are integrated in our build process via git submodules
 - benchmark (https://github.com/google/benchmark)
+- cqqopts (https://github.com/jarro2783/cxxopts.git)
```
Comment (mrks) [inline DEPENDENCIES.md]: typo?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [![Build Status](https://ares.epic.hpi.uni-potsdam.de/jenkins/buildStatus/icon?job=Hyrise/hyrise/master)](https://ares.epic.hpi.uni-potsdam.de/jenkins/blue/organizations/jenkins/Hyrise%2Fhyrise/activity)
-[![Coverage Status](https://hyrise-coverage-badge.herokuapp.com/coverage_badge.svg)](https://ares.epic.hpi.uni-potsdam.de/jenkins/job/Hyrise/job/hyrise/job/master/lastStableBuild/RCov_Report/)
+[![Coverage Status](https://hyrise-coverage-badge.herokuapp.com/coverage_badge.svg)](https://ares.epic.hpi.uni-potsdam.de/jenkins/job/Hyrise/job/hyrise/job/master/lastStableBuild/RCov_20Report/)
```
Comment (mrks) [inline README.md]: The link in the github popup for commits is still broken (see Jenkinsfile)

**Comment by lawben** [comment]
general question: how does this make a -3.1% diff? the diff looks pretty small... are we sure that it is doing the right thing?

File: README.md (hunk 2 above)
Comment (Bensk1) [reply README.md]: Good point, didn't know of that one.

**Comment by Bensk1** [comment]
Maybe because we are excluding test and benchmark code from the coverage calculation now.

**Pull request merged**